modules/stochastic_tools/test/src/postprocessors/OptimizationTestFunction.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "OptimizationTestFunction.h"
+
+registerMooseObject("StochasticToolsTestApp", OptimizationTestFunction);
+
+InputParameters
+OptimizationTestFunction::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+
+  MooseEnum funcs("rosen eggholder");
+  params.addRequiredParam<MooseEnum>("function", funcs, "Type of function to evaluate.");
+  params.addRequiredParam<std::vector<Real>>("x", "Input values.");
+  params.declareControllable("x");
+  return params;
+}
+
+OptimizationTestFunction::OptimizationTestFunction(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _function(getParam<MooseEnum>("function")),
+    _x(getParam<std::vector<Real>>("x"))
+{
+}
+
+PostprocessorValue
+OptimizationTestFunction::getValue() const
+{
+  if (_function == "rosen")
+    return rosen(_x);
+  else if (_function == "eggholder")
+    return eggholder(_x);
+  else
+    paramError("function", "Unknown function: ", _function);
```

Comment:
Should have this check in the constructor, no?

Reply:
It's a MooseEnum, show it should error during parsing. I really will never be hit, anyway.